### ISSUE TITLE
path resolution fails for scoped packages with string as bin value

### ISSUE THREAD

**Issue by knksmith57** [thlorenz/resolve-bin#6]
The [npm `bin` docs](https://docs.npmjs.com/files/package.json#bin) aren't very clear about the intended behavior for [scoped packages](https://docs.npmjs.com/misc/scope) using a string as the value for the `bin` field.

If we [dig a bit](https://github.com/npm/npm/blob/5507d5872b1040e8295c9946467ce5a4c41a7e79/package.json#L70) into the internals, we'll find that [`normalize-package-data`](https://www.npmjs.com/package/normalize-package-data) is used for the heavy lifting when parsing `package.json`.

[PR #69](https://github.com/npm/normalize-package-data/pull/69) [addressed](https://github.com/npm/normalize-package-data/commit/d8a6a0c17ccd173c4d83daab7617e1b9ef981e52) the issue which has been copied below for posterity:
> Installing a scoped module w/ a string bin field would result in a binary named @scope/package, which is impossible to call without an absolute path, which useless to basically everyone. This changes the default to be just package. If it conflicts w/ a non-scoped package, npm@3 will handle the conflict in the usual ways, and the package owner can always explicitly set the bin name to whatever it is they want.

In short, [defaulting the executable name to the literal string value of the package `name` field](https://github.com/thlorenz/resolve-bin/blob/3e9b84e04b690cb27b0e666cc01d9b7d8711d608/index.js#L36) is insufficient.

I think the most future-proof solution to this problem is to use the [`normalize-package-data`](https://www.npmjs.com/package/normalize-package-data) module to ensure any changes landed there (and therefore in `npm` itself) are reflected here.

Thoughts?

Note: for consumers currently blocked by the failed resolution, [manually setting the `opts.executable` option](https://github.com/thlorenz/resolve-bin#parameters) is a solid work around.

**Comment by bendrucker** [comment]
Sure, sharing a dep with npm would be great. PR welcome!
